### ISSUE TITLE
[BUG] nativeTheme.shouldUseDarkColors does not change when the macOS theme is changed

### ISSUE THREAD

**Issue by nathanlesage** [electron/electron#21899]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  * v7.1.10
* **Operating System:**
  * macOS 10.15.2
* **Last Known Working Electron version:**
  * n/a (it worked with Mojave; but since Catalina it does not work anymore)

### Expected Behavior

As soon as `systemPreferences.subscribeNotification('AppleInterfaceThemeChangedNotification', fn)` is triggered (works as expected), `nativeTheme.shouldUseDarkColors` and `systemPreferences.effectiveAppearance` should be set to the correct operating system values, that is, `false`/`light` for light mode and `true`/`dark` for dark mode so that it can be used to determine whether my Electron app should be in dark or light mode.

### Actual Behavior

Both values are always `false`/`light`, no matter how often I switch between the themes in macOS's system preferences.

**Note: `NSRequiresAquaSystemAppearance` is correctly set to `<false/>`, as documented.**

### To Reproduce

Relevant, non-working code ([Link to these lines in repository](https://github.com/Zettlr/Zettlr/blob/0c5718d0b97147567569bf804512784131bbbb19/source/main/providers/appearance-provider.js#L49-L57)):

```javascript
const { systemPreferences, nativeTheme } = require('electron')

// ...

if (process.platform === 'darwin') {
  systemPreferences.subscribeNotification('AppleInterfaceThemeChangedNotification', (event, userInfo) => {
    // Only react to these notifications if the application's own preferences are set to 'system'
    if (this._mode !== 'system') return
    global.log.verbose('Switching to ' + (nativeTheme.shouldUseDarkColors ? 'dark' : 'light') + ' mode')
    global.config.set('darkTheme', nativeTheme.shouldUseDarkColors)
  })
}
```

[Here is a gist](https://gist.github.com/nathanlesage/2017972ed15d4573ba9e879ed259c026) with code you can simply drop into a fresh Electron Fiddle (at the bottom of main.js).

### Additional Information

The problem exists since macOS Catalina; on Mojave it worked perfectly well. The bug has been reported here: https://github.com/Zettlr/Zettlr/issues/362

Before Catalina I used NightOwl for automatic dark mode, but I guess this might not be the problem, as basically everyone was using this … or is this the big mystery?

Thank you very much for building Electron, and thank you for your time! :)

**Comment by MarshallOfSound** [comment]
```
  systemPreferences.subscribeNotification('AppleInterfaceThemeChangedNotification', (event, userInfo) => {
```

This is not the event you want to use, you also probably have a logic flaw, let's run through a few things.

1. If `nativeTheme.themeSource` is set to anything other than "system" then the [`updated`](https://www.electronjs.org/docs/api/native-theme#event-updated) event will not fire.
2. The `AppleInterfaceThemeChangedNotification` is a race condition on Catalina, you should be using the `updated` event I linked above.

TLDR: Change your code to use the `updated` event but be aware of the gotcha in Number 1

**Comment by nathanlesage** [comment]
Thanks a lot for the quick response! :)

Unfortunately, the problem still persists, even using the new event you recommended.

1. `nativeTheme.themeSource` is set to `system`
2. None of the values according to the documentation has changed.

![image](https://user-images.githubusercontent.com/17251683/73202530-88aaeb00-413b-11ea-9447-06ef933f77aa.png)

As you can see, I deliberately switched between light and dark mode in the System Preferences three times, the event (`nativeTheme.on('updated', fn)`) does fire, but none of the underlying values ever changes. And I'm console-logging the values immediately, so there is none of my logic interfering.

Could this have anything to do with having a frameless Window with inset traffic lights, i.e. because of no GUI elements present on the Window itself?

**Comment by MarshallOfSound** [comment]
@nathanlesage Can you try reproduce this in an isolated environment, maybe in a [fiddle](https://electronjs.org/fiddle)?

**Comment by nathanlesage** [comment]
Yes, I just did so: https://gist.github.com/5886e1e4adf4d08ba365451ee5065673

**Comment by nathanlesage** [comment]
Anything new on this one?

**Comment by nathanlesage** [comment]
Just a quick update: The bug persists in Electron 8, with `nativeTheme.shouldUseDarkColors` still always returning false, no matter the OS theme.

**Comment by corysimmons** [comment]
I can confirm this does work in Electron 8.3.4, at least.

```js
let tray

const createWindow = async () => {
	// ...

	const trayPicker = () => {
        if (nativeTheme.shouldUseDarkColors) {
            tray = new Tray('./white.png')
        } else {
            tray = new Tray('./black.png')
        }
    }
    trayPicker() // init
    nativeTheme.on('updated', () => {
        // if someone changes their OS theme, delete then create new icon
        tray.destroy()
        trayPicker()
    })

	// ...
})
```

**Comment by nathanlesage** [comment]
@corysimmons Thanks for commenting, as I can now report there was never a bug, but a misconfiguration, probably introduced by using NightOwl: My *global* setting for dark mode was set to ignore that setting, it was a global default that was misplaced, so it was a problem with Apple's nonexistent documentation on these global defaults …
